src/intrinsics/ecma262/Array.js

```diff
@@ -85,7 +86,12 @@ export default function (realm: Realm): NativeFunctionValue {
         // c. Let intLen be 1.
         intLen = 1;
       } else { // 7. Else,
-        len = len.throwIfNotConcreteNumber();
+        // len = recoverableError(() => len.throwIfNotConcreteNumber(), ()=>new NumberValue(realm, 0));
+        len = recoverableError(
+          realm,
+          ()=>len.throwIfNotConcreteNumber(),
+          ()=>new NumberValue(realm, 0));
+
```

Comment:
Hmm. First you write the entire statement in one line, but in the form of a comment. Then you repeat it verbatim, but you break it up into multiple lines.

I'm also not thrilled with the helper function. I'd rather see something like "if (something is bad && reportError("Hey, this thing is bad because it's BAD") === "recover") { do some recovery thing };".